geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearWithConcurrentOperationsDUnitTest.java

```diff
@@ -0,0 +1,525 @@
+   * - Clears the Partition Region continuously (once every 2 seconds for 10 seconds).
+   * - Asserts that, after the clears are finished, the Partition Region is empty on all members.
+   */
+  @Test
+  @TestCaseName(TEST_CASE_NAME)
+  @Parameters(method = "vmAndRegionTypes")
+  public void clearWithConcurrentGetsShouldWorkCorrectly(TestVM coordinatorVM,
+      RegionShortcut regionShortcut) throws InterruptedException {
+    final int entries = 10000;
+    parametrizedSetup(regionShortcut);
+    populateRegion(accessor, entries, asList(accessor, server1, server2));
+
+    // Let all VMs continuously execute gets for 10 seconds.
+    List<AsyncInvocation<Void>> asyncInvocationList = Arrays.asList(
+        server1.invokeAsync(() -> executeGets(entries, 10)),
+        server2.invokeAsync(() -> executeGets(entries, 10)),
+        accessor.invokeAsync(() -> executeGets(entries, 10)));
+
+    // Clear the region every 2 seconds for 10 seconds.
+    getVM(coordinatorVM.vmNumber).invoke(() -> executeClears(10, 2));
+
+    // Let asyncInvocations finish.
+    for (AsyncInvocation<Void> asyncInvocation : asyncInvocationList) {
+      asyncInvocation.await();
+    }
+
+    // Assert Region is empty across VMs.
+    assertRegionIsEmpty(asList(accessor, server1, server1));
+  }
+
+  /**
+   * The test does the following (the clear coordinator is chosen through parameters):
+   * - Launches one thread per VM to continuously execute puts for 15 seconds.
+   * - Clears the Partition Region continuously (once every 3 seconds for 15 seconds).
+   * - Asserts that, after the clears are finished, the Region data is consistent across members.
+   */
+  @Test
+  @TestCaseName(TEST_CASE_NAME)
+  @Parameters(method = "vmAndRegionTypes")
+  public void clearWithConcurrentPutsShouldWorkCorrectly(TestVM coordinatorVM,
+      RegionShortcut regionShortcut) throws InterruptedException {
+    final int entries = 5000;
+    parametrizedSetup(regionShortcut);
+
+    // Let all VMs continuously execute puts for 15 seconds.
+    List<AsyncInvocation<Void>> asyncInvocationList = Arrays.asList(
+        server1.invokeAsync(() -> executePuts(entries, 15)),
+        server2.invokeAsync(() -> executePuts(entries, 15)),
+        accessor.invokeAsync(() -> executePuts(entries, 15)));
+
+    // Clear the region every 3 seconds for 15 seconds.
+    getVM(coordinatorVM.vmNumber).invoke(() -> executeClears(15, 3));
+
+    // Let asyncInvocations finish.
+    for (AsyncInvocation<Void> asyncInvocation : asyncInvocationList) {
+      asyncInvocation.await();
+    }
+
+    // Assert Region Data is consistent across members.
+    assertRegionDataConsistency(asList(accessor, server1, server2));
+  }
+
+  /**
+   * The test does the following (the clear coordinator is chosen through parameters):
+   * - Launches threads to continuously execute puts and gets for 60 seconds.
+   * - Clears the Partition Region continuously (once every 10 seconds for 60 seconds).
```

Comment:
No need to specify time; if someone changes the time and doesn't update it, this will become incorrect.

Reply:
Done!.